nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,109 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.jersey.core.util.MultivaluedMapImpl;
 
 @RunWith(FeaturesRunner.class)
-@Features(CoreFeature.class)
+@Features(RestServerFeature.class)
 @RepositoryConfig(cleanup = Granularity.METHOD)
 @Deploy("org.nuxeo.ecm.platform.video:OSGI-INF/core-types-contrib.xml")
 @Deploy("org.nuxeo.ecm.platform.audio.core:OSGI-INF/core-types-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.tag:OSGI-INF/faceted-tag-service-core-types.xml")
 @Deploy("org.nuxeo.ecm.platform.picture.core:OSGI-INF/picture-schemas-contrib.xml")
-@Deploy("org.nuxeo.ecm.platform.types:OSGI-INF/subtypes-enricher-contrib.xml")
 @Deploy("org.nuxeo.compound.documents")
-public class TestCompoundDocument extends AbstractJsonWriterTest.Local<DocumentModelJsonWriter, DocumentModel> {
+public class TestCompoundDocument {
 
     @Inject
-    protected CoreSession session;
-    
-    public TestCompoundDocument() {
-        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected HttpClientTestRule httpClientRule;
+
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    protected Set<String> subtypes = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+
+    @Before
+    public void before() {
+        httpClientRule = new HttpClientTestRule.Builder().credentials("Administrator", "Administrator")
+                                                         .url(String.format("http://localhost:%d/api/v1",
+                                                                 servletContainerFeature.getPort()))
+                                                         .build();
+        httpClientRule.starting();
     }
 
+    @After
+    public void after() {
+        httpClientRule.finished();
+    }
+
+    // Avoid useless warning for not finding dclistener
+    @Deploy("org.nuxeo.compound.documents:OSGI-INF/disable-dclistener-contrib.xml")
     @Test
     public void testCompoundDocument() throws IOException {
-        DocumentModel doc = session.createDocument(session.createDocumentModel("/", "test", "CompoundDocument"));
-        assertTrue(doc.getFacets().contains("CompoundDocument"));
-        RenderingContext ctx = RenderingContext.CtxBuilder.enrichDoc("subtypes").get();
-        JsonAssert json = jsonAssert(doc, ctx);
-        json = json.has("contextParameters").isObject();
-        json.properties(1);
-        json = json.has("subtypes").isArray();
-        json.childrenContains("type", "File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+        var formData = new MultivaluedMapImpl();
+        formData.putSingle("doctype", "CompoundDocument");
+        formData.putSingle("name", "myCompoundDocument");
+        try (CloseableClientResponse response = httpClientRule.post("/path/", formData)) {
+            assertEquals(SC_SEE_OTHER, response.getStatus());
+        }
```

Comment:
May the JSON payload be more interesting for document creation? wdyt, cc @troger?

Reply:
Agree, I would prefer that we POST the doc JSON, as the UI will do it.